logstash-core/src/main/java/org/logstash/health/PipelineIndicator.java

```diff
@@ -0,0 +1,171 @@
+            this.pipelineDetailsProvider = pipelineDetailsProvider;
+        }
+
+        public Details get() {
+            return this.pipelineDetailsProvider.pipelineDetails(pipelineId);
+        }
+    }
+
+    static class StatusProbe implements Probe<Details> {
+        static final Impact.Builder NOT_PROCESSING = Impact.builder()
+                .withId(impactId("not_processing"))
+                .withDescription("the pipeline is not currently processing")
+                .withAdditionalImpactArea(ImpactArea.PIPELINE_EXECUTION);
+
+        @Override
+        public Analysis analyze(final Details details) {
+            switch (details.getStatus().getState()) {
+                case LOADING:
+                    return Analysis.builder()
+                            .withStatus(YELLOW)
+                            .withDiagnosis(db -> db
+                                    .withId(diagnosisId("loading"))
+                                    .withCause("pipeline is loading")
+                                    .withAction("if pipeline does not come up quickly, you may need to check the logs to see if it is stalled"))
+                            .withImpact(NOT_PROCESSING.withSeverity(1).withDescription("pipeline is loading").build())
+                            .build();
+                case RUNNING:
+                    return Analysis.builder()
+                            .withStatus(GREEN)
+                            .build();
+                case FINISHED:
+                    return Analysis.builder()
+                            .withStatus(YELLOW)
+                            .withDiagnosis(db -> db
+                                    .withId(diagnosisId("finished"))
+                                    .withCause("pipeline has finished running because its inputs have been closed and events have been processed")
+                                    .withAction("if you expect this pipeline to run indefinitely, you will need to configure its inputs to continue receiving or fetching events"))
+                            .withImpact(NOT_PROCESSING.withSeverity(10).withDescription("pipeline has finished running").build())
+                            .build();
+                case TERMINATED:
+                    return Analysis.builder()
+                            .withStatus(RED)
+                            .withDiagnosis(db -> db
+                                    .withId(diagnosisId("terminated"))
+                                    .withCause("pipeline is not running, likely because it has encountered an error")
+                                    .withAction("view logs to determine the cause of abnormal pipeline shutdown"))
+                            .withImpact(NOT_PROCESSING.withSeverity(1).build())
+                            .build();
+                case UNKNOWN:
+                default:
+                    return Analysis.builder()
+                            .withStatus(YELLOW)
```

Comment:
This is the catch-all for when we don't _know_ enough to report on it. We should improve the underlying state tracking _so that_ there are ever-fewer cases in which we cannot ascertain the state from outside the pipeline. 

I'll resolve the crash, which as you speculated is related to my refactor to make `status.state` instead of `run_state`, which is what gives us the room for things like `status.goal`, `status.last_change`, and others in the future.

Reply:
I _think_ that our concerns are better met when inverting the `green` and `unknown` statuses so that `unknown` is a more-degraded state and can bubble up, which is now addressed in this PR.

Let's file separate work to have an actual state machine that makes observing these edge-cases more straight-forward, which should eliminate the crash-in-register unknown state.

Reply:
> our concerns are better met when inverting the green and unknown statuses so that unknown is a more-degraded state and can bubble up, which is now addressed in this PR.

Agreed. Thank you!